File: frontend/demo/component/dialog/dialog-closing.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/button';
import '@vaadin/dialog';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { dialogFooterRenderer, dialogRenderer } from '@vaadin/dialog/lit.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('dialog-closing')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private dialogOpened = false;

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-dialog
        header-title="System maintenance"
        .opened="${this.dialogOpened}"
        @closed="${() => {
          this.dialogOpened = false;
        }}"
        ${dialogRenderer(
          () => html`
            <p style="max-width: 300px">
              System maintenance will begin at 3 PM. It is schedule to conclude at 5PM. We apologise
              for any inconvenience.
            </p>
          `,
          []
        )}
        ${dialogFooterRenderer(
          () => html`<vaadin-button @click="${this.close}">Close</vaadin-button>`,
          []
        )}
      ></vaadin-dialog>
      <!-- end::snippet[] -->

      <vaadin-button @click="${this.open}">Show dialog</vaadin-button>
    `;
  }

  private open() {
    this.dialogOpened = true;
  }

  private close() {
    this.dialogOpened = false;
  }
}

File: src/main/java/com/vaadin/demo/component/dialog/DialogClosing.java
package com.vaadin.demo.component.dialog;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.dialog.Dialog;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.Paragraph;
import com.vaadin.flow.router.Route;

@Route("dialog-closing")
public class DialogClosing extends Div {

    public DialogClosing() {
        Dialog dialog = new Dialog();

        dialog.setHeaderTitle("System maintenance");

        Paragraph text = new Paragraph(
                "System maintenance will begin at 3 PM. It is schedule to conclude at 5PM. We apologize for any inconvenience.");
        text.getStyle().set("max-width", "300px");
        dialog.add(text);

        // tag::snippet[]
        Button closeButton = new Button("Close");
        closeButton.addClickListener(e -> dialog.close());
        // end::snippet[]
        dialog.getFooter().add(closeButton);

        Button button = new Button("Show dialog", e -> dialog.open());
        add(dialog, button);
    }

    public static class Exporter extends DemoExporter<DialogClosing> { // hidden-source-line
    } // hidden-source-line
}
